Dockerfile

```diff
@@ -127,18 +127,18 @@ WORKDIR /home/${USERNAME}
 ENV PATH="${PATH}:/home/${USERNAME}/.local/bin"
 
 # Upgrade pip and install packages.
-RUN python3.10 -m pip install --no-cache-dir --upgrade pip setuptools pathtools promise pybind11 omegaconf
+RUN python3.10 -m pip install --no-cache-dir --upgrade pip setuptools==69.5.1 pathtools promise pybind11 omegaconf
 
 # Install pytorch and submodules
 # echo "${CUDA_VERSION}" | sed 's/.$//' | tr -d '.' -- CUDA_VERSION -> delete last digit -> delete all '.'
 RUN CUDA_VER=$(echo "${CUDA_VERSION}" | sed 's/.$//' | tr -d '.') && python3.10 -m pip install --no-cache-dir \
-    torch==2.0.1+cu${CUDA_VER} \
-    torchvision==0.15.2+cu${CUDA_VER} \
+    torch==2.1.2+cu${CUDA_VER} \
+    torchvision==0.16.2+cu${CUDA_VER} \
         --extra-index-url https://download.pytorch.org/whl/cu${CUDA_VER}
 
 # Install tiny-cuda-nn (we need to set the target architectures as environment variable first).
 ENV TCNN_CUDA_ARCHITECTURES=${CUDA_ARCHITECTURES}
-RUN python3.10 -m pip install --no-cache-dir git+https://github.com/NVlabs/tiny-cuda-nn.git@v1.6#subdirectory=bindings/torch
+RUN python3.10 -m pip install --no-cache-dir git+https://github.com/NVlabs/tiny-cuda-nn.git#subdirectory=bindings/torch
```

Comment:
why remove this version pin?  instead of updating it to whatever commit for this PR?  tiny-cuda-nn is fairly volatile so removing the version pin makes this Dockerfile more like a cookbook than a reproducible spec

Reply:
As written in the PR description: to my experience the latest version (1.6) is not compatible with the requested pytorch version. I agree that pinned versions are always the better way to go but could not build the docker image with 1.6 and torch 2.1.2. if you have another solution feel free to share.

Reply:
@Zunhammer you can pip-install at a specific commit like this: https://stackoverflow.com/a/13754517

or in a Dockerfile, you can also do arbitrary commands like just clone / checkout the repo at a commit and then `pip install .`